lib/pages/planning_page/manage_budget_page.dart

```diff
@@ -0,0 +1,162 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../../constants/style.dart';
+import '../../model/budget.dart';
+import '../../model/category_transaction.dart';
+import 'widget/budget_category_selector.dart';
+import '../../../providers/categories_provider.dart';
+import '../../../providers/budgets_provider.dart';
+
+class ManageBudgetPage extends ConsumerStatefulWidget {
+  final Function() onRefreshBudgets;
+  const ManageBudgetPage({required this.onRefreshBudgets, super.key});
+
+  @override
+  ConsumerState<ManageBudgetPage> createState() => _ManageBudgetPageState();
+}
+
+class _ManageBudgetPageState extends ConsumerState<ManageBudgetPage> {
+  List<CategoryTransaction> categories = [];
+  List<Budget> budgets = [];
+  List<Budget> deletedBudgets = [];
+
+  void _loadCategories() async {
+    categories = await ref.read(categoriesProvider.notifier).getCategories();
+    budgets = await ref.read(budgetsProvider.notifier).getBudgets();
+    setState(() {});
+  }
+
+  void updateBudget(Budget updatedBudget, int index) {
+    setState(() {
+      deletedBudgets.add(budgets[index]);
+      budgets[index] = updatedBudget;
+    });
+  }
+
+  void deleteBudget(Budget removedBudget, int index) {
+    setState(() {
+      budgets.removeAt(index);
+      deletedBudgets.add(removedBudget);
+    });
+  }
+
+  @override
+  void initState() {
+    _loadCategories();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Padding(
+            padding: const EdgeInsets.all(15),
+            child: Text("Select the categories to create your budget",
+                style: Theme.of(context).textTheme.titleLarge)),
+        const Padding(
+            padding: EdgeInsets.symmetric(horizontal: 15),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  'Category',
+                  textAlign: TextAlign.left,
+                ),
+                Text(
+                  'Amount',
+                  textAlign: TextAlign.right,
+                ),
+              ],
+            )),
+        Expanded(
+            child: Column(children: [
+          ListView.builder(
```

Comment:
Below the list of categories we should have a text that explains to the user that if they want to delete a budget they should swipe it left or right.

Reply:
Added